### PR TITLE
Update validation rule in appstream stack resource

### LINKED ISSUE #27743: [Bug]: appstream stack redirect url should have a maximum value of 1000
### Terraform Core Version

1.3.0

### AWS Provider Version

since 3.56.0

### Affected Resource(s)

* aws_appstream_stack

### Expected Behavior

The maximum length of the redirect url is 1000 (from docs: https://docs.aws.amazon.com/appstream2/latest/APIReference/API_CreateStack.html#AppStream2-CreateStack-request-RedirectURL)

### Actual Behavior

It's capped at 100 characters

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

n/a

### Steps to Reproduce

n/a

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by pierskarsenbarg**
### Description

Increase max length of redirectUrl in appstream stack resource. Docs specify that max length is 1000: https://docs.aws.amazon.com/appstream2/latest/APIReference/API_CreateStack.html#AppStream2-CreateStack-request-RedirectURL

### Relations

Fixes #27743

### Output from Acceptance Testing

```
$ make testacc TESTS=TestAccXXX PKG=ec2

...
```

### CODE REVIEW

File: internal/service/appstream/stack.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func ResourceStack() *schema.Resource {
 						"settings_group": {
 							Type:         schema.TypeString,
 							Optional:     true,
-							ValidateFunc: validation.StringLenBetween(0, 100),
+							ValidateFunc: validation.StringLenBetween(0, 1000),
```
Comment (jar-b) [inline internal/service/appstream/stack.go:78]: This limit on `settings_group` is correctly set to 100.
https://docs.aws.amazon.com/appstream2/latest/APIReference/API_ApplicationSettings.html#AppStream2-Type-ApplicationSettings-SettingsGroup

I think maybe you meant to change this ValidateFunc?

https://github.com/hashicorp/terraform-provider-aws/blob/f23a9c4ae983e7ca765397474e342db26059a636/internal/service/appstream/stack.go#L128-L133

Comment (pierskarsenbarg) [reply internal/service/appstream/stack.go]: Well, that's embarrassing 🤦

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
$ make testacc PKG=appstream TESTS=TestAccAppStreamStack_ ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/appstream/... -v -count 1 -parallel 2 -run='TestAccAppStreamStack_'  -timeout 180m
=== RUN   TestAccAppStreamStack_basic
=== PAUSE TestAccAppStreamStack_basic
=== RUN   TestAccAppStreamStack_disappears
=== PAUSE TestAccAppStreamStack_disappears
=== RUN   TestAccAppStreamStack_complete
=== PAUSE TestAccAppStreamStack_complete
=== RUN   TestAccAppStreamStack_applicationSettings_basic
=== PAUSE TestAccAppStreamStack_applicationSettings_basic
=== RUN   TestAccAppStreamStack_applicationSettings_removeFromEnabled
=== PAUSE TestAccAppStreamStack_applicationSettings_removeFromEnabled
=== RUN   TestAccAppStreamStack_applicationSettings_removeFromDisabled
=== PAUSE TestAccAppStreamStack_applicationSettings_removeFromDisabled
=== RUN   TestAccAppStreamStack_withTags
=== PAUSE TestAccAppStreamStack_withTags
=== CONT  TestAccAppStreamStack_basic
=== CONT  TestAccAppStreamStack_applicationSettings_removeFromEnabled
--- PASS: TestAccAppStreamStack_basic (17.36s)
=== CONT  TestAccAppStreamStack_complete
--- PASS: TestAccAppStreamStack_applicationSettings_removeFromEnabled (28.63s)
=== CONT  TestAccAppStreamStack_applicationSettings_basic
--- PASS: TestAccAppStreamStack_complete (29.38s)
=== CONT  TestAccAppStreamStack_disappears
--- PASS: TestAccAppStreamStack_disappears (12.45s)
=== CONT  TestAccAppStreamStack_withTags
--- PASS: TestAccAppStreamStack_applicationSettings_basic (43.13s)
=== CONT  TestAccAppStreamStack_applicationSettings_removeFromDisabled
--- PASS: TestAccAppStreamStack_withTags (29.26s)
--- PASS: TestAccAppStreamStack_applicationSettings_removeFromDisabled (21.03s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/appstream  95.417s
```

**Comment by jar-b** [comment]
Nice find, @pierskarsenbarg! Thanks for your contribution 👏

**Pull request merged**